### ISSUE TITLE
Make required error more accessible

### ISSUE THREAD

**Issue by Looooooka** [gregsdennis/json-everything#371]
**Environment**
- Nuget Package: JsonSchema.Net
- Nuget Version: 4.0.0-beta2
- OS: Windows
- .Net Target: net6.0

The evaluation details aren't as useful as they could be. Specific scenario:
Current .NET validation logic for  most Javascript/UI frameworks supports server side validation errors using a response with an Errors Dictionary<String,String[]>
Meaning we only need to return 

```json
{ 
"Errors": { 
"SomeProperty": 
[ "Required Property missing" ] ,
"SomeOtherProperty": 
[ 
"Min length 6 required" ,
"Some other error"
] 
}
}
```
The current evaluation details make generating this when a required field is missing quite difficult:
```json
{
  "valid": false,
  "details": [
    {
      "valid": false,
      "evaluationPath": "",
      "schemaLocation": "https://json-everything.net/dc09ba01a7",
      "instanceLocation": "",
      "errors": {
        "required": "Required properties [\"Message\"] were not present"
      }
    },
    {
      "valid": true,
      "evaluationPath": "/properties/Message2",
      "schemaLocation": "https://json-everything.net/dc09ba01a7#/properties/Message2",
      "instanceLocation": "/Message2"
    },
    {
      "valid": false,
      "evaluationPath": "/properties/ComplexObject",
      "schemaLocation": "https://json-everything.net/dc09ba01a7#/properties/ComplexObject",
      "instanceLocation": "/ComplexObject"
    },
    {
      "valid": false,
      "evaluationPath": "/properties/ComplexObject/properties/Name",
      "schemaLocation": "https://json-everything.net/dc09ba01a7#/properties/ComplexObject/properties/Name",
      "instanceLocation": "/ComplexObject/Name",
      "errors": {
        "minLength": "Value is not longer than or equal to 10 characters"
      }
    }
  ]
}
```
It's almost impossible because figuring out that Message is missing requires parsing the errors->required message and looking for for the missing properties inside the message, which in the future could very well be in another language.
Personally I'd just add that as a separate detail with the evaluationPath of "/Properties/Message" and add "required" under errors.
So basically get rid of the first item and replace it with(for each missing property) or leave it and still add the required error for the missing Message property:

```json
{
      "valid": false,
      "evaluationPath": "/properties/Message",
      "schemaLocation": "https://json-everything.net/dc09ba01a7#/properties/Message",
      "instanceLocation": "/Message",
      "errors": {
        "required": "Required property not present"
}
```

**Comment by gregsdennis** [comment]
This is an interesting idea, but there are a few issues with it.

- `/properties/Message` isn't raising the error, `/required` is
- `/Message` isn't a valid location in the instance (becuase it's missing, hence the error)

The purposes of `evaluationPath` and `instanceLocation` are to indicate where in the schema and in the instance (respectively) the problem occurred.

`properties` is saying, "If this object has a `Message` property, its value must be _X_."  But it doesn't have that property so, it just continues.

`required` is saying, "This object should have a `Message` property but doesn't."

`properties` doesn't have a problem with the instance, but `required` does.  Thus the error needs to come from `required`.

This is how the specification regards output.  if you want to join the conversation on that, please [open an issue in the spec repo](https://github.com/json-schema-org/json-schema-spec/issues) or a [discussion in the JSON Schema community repo](https://github.com/orgs/json-schema-org/discussions).

---

If you're having a problem parsing the error message, I suggest you simply [create your own error message](https://json-everything.net/json-schema#customizing-error-messages) that's easier to parse.
